operator/cmd/supervisor/supervisor.go

```diff
@@ -0,0 +1,159 @@
+package supervisor
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"os/signal"
+	"time"
+
+	"github.com/redpanda-data/redpanda-operator/pkg/otelutil/log"
+	"github.com/spf13/cobra"
+	"golang.org/x/time/rate"
+)
+
+func Command() *cobra.Command {
+	var retryBurst int
+	var retryRate time.Duration
+	var gracePeriod time.Duration
+
+	cmd := &cobra.Command{
+		Use:     "supervisor command...",
+		Example: "supervisor --burst-rate=1m -- redpanda-operator sidecar --redpanda-yaml path/on/disk",
+		Short: `supervisor is a stand in for Kubernetes' Sidecar feature.
+It continuously re-runs the provided command, forwarding signals, in a loop until SIGINT is received.
+Subprocess retries are rate limited with token bucket that may be tuned via the --retry-burst and --retry-rate flags.
+The default retry parameters are tuned for a a long running process, such as the sidecar subcommand.`,
+		Args: cobra.MinimumNArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			ctx, cancel := signal.NotifyContext(cmd.Context(), os.Interrupt)
+			defer cancel()
+
+			signals := make(chan os.Signal, 16)
+
+			signal.Notify(signals)
```

Comment:
How does the internal signal notification work with multiple handlers like this? Is it deterministic whether the NotifyContext above gets called as the handler v this notification channel?